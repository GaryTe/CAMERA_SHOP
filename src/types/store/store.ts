import { store } from '../../store/store/store';
import { listCamera, CardProduct, Reviews, PromoProduct, Review } from '../types-response/types-response';

export type Cameras = {
  cameras: listCamera | null;
  modifiedListProduct: listCamera | null;
}

export type CardCamera = {
  dataCamera: CardProduct | null;
}

export type SimilaCameras = {
  dataSimilaCameras: listCamera | null;
  sortListCamerra: listCamera | null;
  counterMin: number;
  counterMax: number;
}

export type ReviewsCamera = {
  dataReviews: Reviews | Array<Review>;
  resultSortReviews: Reviews | null;
  reviews: Reviews | null;
  counter: number;
}

export type PromoCamera = {
  dataPromoCamera: PromoProduct | null;
}

export type Cart = {
  dataProduct: listCamera | Array<CardProduct>;
  loading: boolean;
}

export type RootState = ReturnType<typeof store.getState>;
