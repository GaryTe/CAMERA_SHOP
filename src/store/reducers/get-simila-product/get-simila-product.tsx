import { createSlice } from '@reduxjs/toolkit';
import { fetchDataSimilarProduct } from '../../api-actions/api-actions';
import { SimilaCameras } from '../../../types/store/store';
import { NameSpace } from '../../../const/const';
import { listProduct } from '../../../util/mocks';

const initialState: SimilaCameras = {
  dataSimilaCameras: null,
  sortListCamerra: null,
  counterMin: 0,
  counterMax: 3
};

export const getSimilaProduct = createSlice({
  name: NameSpace.SimilarProduct,
  initialState,
  reducers: {
    showMoreCards: (state) => {
      if(state.dataSimilaCameras === null) {return;}
      state.counterMin += 3;
      state.counterMax += 3;
      state.sortListCamerra = state.dataSimilaCameras;
    },
    showFewerCards: (state) => {
      if(state.dataSimilaCameras === null) {return;}
      state.counterMin -= 3;
      state.counterMax -= 3;
      state.sortListCamerra = state.dataSimilaCameras;
    },
    functionForTesting: (state) => {
      state.sortListCamerra = listProduct();
      state.dataSimilaCameras = listProduct();
    }
  },
  extraReducers(builder) {
    builder
      .addCase(fetchDataSimilarProduct.fulfilled, (state, action) => {
        state.dataSimilaCameras = action.payload;
        state.sortListCamerra = state.dataSimilaCameras;
      });
  }
});

export const {showMoreCards, showFewerCards, functionForTesting} = getSimilaProduct.actions;
