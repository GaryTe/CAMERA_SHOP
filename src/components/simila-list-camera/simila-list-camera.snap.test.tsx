import {render} from '@testing-library/react';
import {Provider} from 'react-redux';
import {useState} from 'react';
import SimilaListCamera from './simila-list-camera';
import { functionForTesting } from '../../store/reducers/get-simila-product/get-simila-product';
import { store } from '../../store/store/store';

const mockUseState = jest.fn(() => useState(false));

describe('Component: SimilaListCamera', () => {
  it('should render SimilaListCamera', () => {
    store.dispatch(functionForTesting);

    const {container} = render(
      <Provider store={store}>
        <SimilaListCamera
          cameraId={'1'}
          onSetMadalWindowAddProduct={mockUseState}
        />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
