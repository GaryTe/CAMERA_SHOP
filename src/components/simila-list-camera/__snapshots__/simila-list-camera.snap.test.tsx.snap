// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SimilaListCamera should render SimilaListCamera 1`] = `
<div>
  <div
    class="page-content__section"
  >
    <section
      class="product-similar"
    >
      <div
        class="container"
      >
        <h2
          class="title title--h3"
        >
          Похожие товары
        </h2>
        <div
          class="product-similar__slider"
          style="overflow: hidden; max-width: 560px;"
        >
          <div
            class="product-similar__slider-list"
            style="transform: translateX(0px); transition: translate; transition-property: transform; transition-duration: 400ms; transition-timing-function: ease-in-out;"
          >
            ;
          </div>
          <button
            aria-label="Предыдущий слайд"
            class="slider-controls slider-controls--prev"
            disabled=""
            type="button"
          >
            <svg
              aria-hidden="true"
              height="12"
              width="7"
            >
              <use
                xlink:href="#icon-arrow"
              />
            </svg>
          </button>
          <button
            aria-label="Следующий слайд"
            class="slider-controls slider-controls--next"
            type="button"
          >
            <svg
              aria-hidden="true"
              height="12"
              width="7"
            >
              <use
                xlink:href="#icon-arrow"
              />
            </svg>
          </button>
        </div>
      </div>
    </section>
  </div>
</div>
`;
